crates/ol/stf/src/account_processing.rs

```diff
@@ -0,0 +1,140 @@
+//! Account-specific interaction handling, such as messages.
+
+use strata_acct_types::{AccountId, AcctError, MsgPayload};
+use strata_codec::encode_to_vec;
+use strata_ledger_types::{
+    AccountTypeState, Coin, IAccountState, ISnarkAccountState, StateAccessor,
+};
+use strata_msg_fmt::{Msg, MsgRef};
+use strata_ol_chain_types_new::{OLLog, SimpleWithdrawalIntentLogData};
+use strata_ol_msg_types::{OLMessageExt, WITHDRAWAL_MSG_TYPE_ID};
+use strata_snark_acct_types::MessageEntry;
+
+use crate::{
+    constants::{BRIDGE_GATEWAY_ACCT_ID, BRIDGE_GATEWAY_ACCT_SERIAL},
+    context::BasicExecContext,
+    errors::{ExecError, ExecResult},
+    output::OutputCtx,
+};
+
+/// Processes a message by delivering it to its destination, which might involve
+/// touching the ledger state.
+///
+/// This takes a [`EpochContext`] because messages can be issued both in regular
+/// block processing and at epoch sealing.
+pub(crate) fn process_message<S: StateAccessor>(
+    state: &mut S,
+    sender: AccountId,
+    target: AccountId,
+    msg: MsgPayload,
+    context: &BasicExecContext<'_>,
+) -> ExecResult<()> {
+    match target {
+        // Bridge gateway messages.
+        BRIDGE_GATEWAY_ACCT_ID => {
+            handle_bridge_gateway_message(state, sender, msg, context)?;
+        }
+
+        // Any other address we assume is a ledger account, so we have to look it up.
+        _ => {
+            // TODO adapt the account state traits to make this all more
+            // amendable to avoiding copies/clones
+
+            // Make a copy of the account state.  I don't love this.
+            let Some(mut acct_state) = state.get_account_state(target)?.cloned() else {
+                // If we don't find it then we can just ignore it.
+                // TODO do something with the funds we're throwing away by doing this
```

Comment:
This `TODO` is dangerous enough (loss of funds) that I think deserves a blocking review comment.